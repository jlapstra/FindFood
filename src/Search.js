import React, { Component } from 'react';
import $ from 'jquery';
import './Search.css';

import Button from 'react-bootstrap/Button';
import InputGroup from 'react-bootstrap/InputGroup';
import FormControl from 'react-bootstrap/FormControl';

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = { restaurantList: []};
    this.search = this.search.bind(this);
    this.updateState = this.updateState.bind(this);
  }
  render() {
    return (
      <div className="Search">
        <InputGroup>
          <InputGroup.Prepend>
            <InputGroup.Text>Please enter your desired city:</InputGroup.Text>
          </InputGroup.Prepend>
          <FormControl id="input" size="lg" />
        </InputGroup>
        <br />
        <Button id="search" variant="success" onClick={this.search}>Search!</Button>
      </div>
    );
  }
  updateState(restaurantList) {
    this.props.search(restaurantList);
  }
  search() {
    var u = 'https://opentable.herokuapp.com/api/restaurants';
    var restaurantList = [];
    var that = this;
    console.log($('#input').val());

    $.ajax( {
      type: 'GET',
      url: u,
      data: {city: $('#input').val() },
      success: function(response) {
        if (response.restaurants.length === 0) {
          alert("City not found, please try again");
        }
        $.each(response.restaurants, function(index, element) {
          restaurantList.push({
            name: element.name,
            price: element.price,
            address: element.address,
            img: element.image_url,
            info: element.reserve_url
          });
        });
        that.updateState(restaurantList);
      },
      error: function(xhr) {
        alert(xhr.responseJSON.error);
      },
      dataType: 'json'
    });
  }
}

export default Search;
