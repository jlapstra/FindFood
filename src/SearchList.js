import React, { Component } from 'react';
import './SearchList.css';

import Restaurant from './Restaurant';
import Search from './Search';


class SearchList extends Component {
  constructor(props) {
    super(props);
    this.state = { restaurantList: [] };
    this.search = this.search.bind(this);
  }
  search(rList) {
    this.setState({ restaurantList: rList });
    this.forceUpdate();
  }
  render() {
    return (
      <div className='SearchList'>
        <Search search={this.search} />
        { this.renderResults() }
      </div>
    );
  }
  renderResults() {
    try {
      return this.state.restaurantList.map((r, index) => (
        <Restaurant key={r.name+index} name={r.name} price={r.price} address={r.address} img={r.img} info={r.info}/>
      ));
    }
    catch(err) {
      console.log(err);
      return
    }
  }
}

export default SearchList;
