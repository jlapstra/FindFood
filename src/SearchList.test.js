import React from 'react';
import ReactDOM from 'react-dom';
import SearchList from './SearchList';
import Search from './Search';

import { mount, shallow } from 'enzyme';

const rList = [
  {
    name: 'Jenna',
    price: 2,
    address: '1 Queen',
    img: undefined,
    info: undefined
  },
  {
    name: 'Troy',
    price: 2,
    address: '3 Front',
    img: undefined,
    info: undefined
  }
];

const jsontest = {"total_entries":2,"per_page":25,"current_page":1,"restaurants":[{"id":21307,"name":"Jenna","address":"1 Queen","city":"Toronto","state":"ON","area":"Toronto / SW Ontario","postal_code":"M4V 2L1","country":"CA","phone":"4169618011","lat":43.68207,"lng":-79.40041,"price":2},{"id":82957,"name":"Troy","address":"3 Front","city":"Toronto","state":"ON","area":"Toronto / SW Ontario","postal_code":"M5E 1B3","country":"CA","phone":"4169610601x","lat":43.648033,"lng":-79.374377,"price": 2}]};


describe('SearchList Component', () => {
  it('renders appropriately with restaurantList given', () => {
    const component = mount(<SearchList />);

    component.setState({restaurantList: rList});

    expect(component).toMatchSnapshot();

    component.unmount();
  });
});

describe('Search Component', () => {
  it('should populate data once city selected', () => {
    const component = mount(<SearchList/>);
    const $ = require('jquery');

    jest.unmock('jquery');

    $.ajax = jest.fn().mockImplementation(function(x) {
      console.log(x.success);
      x.success(jsontest); 
    });

    component.find('FormControl#input').simulate('change', {
        target: { value: 'Chicago' }
      });

    component
      .find('Button#search')
      .simulate('click');

    expect(component.state('restaurantList')).toEqual(rList);

    component.unmount();

    jest.resetModules();
  });
});
