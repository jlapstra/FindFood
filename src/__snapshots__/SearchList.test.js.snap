// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchList Component renders appropriately with restaurantList given 1`] = `
<SearchList>
  <div
    className="SearchList"
  >
    <Search
      search={[Function]}
    >
      <div
        className="Search"
      >
        <Bootstrap(InputGroup)>
          <InputGroup
            bsPrefix="input-group"
          >
            <div
              className="input-group"
            >
              <InputGroupPrepend>
                <div
                  className="input-group-prepend"
                >
                  <InputGroupText>
                    <span
                      className="input-group-text"
                    >
                      Please enter your desired city:
                    </span>
                  </InputGroupText>
                </div>
              </InputGroupPrepend>
              <FormControl
                id="input"
                size="lg"
              >
                <input
                  className="form-control form-control-lg"
                  id="input"
                />
              </FormControl>
            </div>
          </InputGroup>
        </Bootstrap(InputGroup)>
        <br />
        <Button
          active={false}
          disabled={false}
          id="search"
          onClick={[Function]}
          type="button"
          variant="success"
        >
          <button
            className="btn btn-success"
            disabled={false}
            id="search"
            onClick={[Function]}
            type="button"
          >
            Search!
          </button>
        </Button>
      </div>
    </Search>
    <Restaurant
      address="1 Queen"
      key="Jenna0"
      name="Jenna"
      price={2}
    >
      <div
        className="Restaurant"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <Card
          body={false}
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <div
            className="card"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <CardImg
              variant="top"
            >
              <img
                className="card-img-top"
              />
            </CardImg>
            <CardBody>
              <div
                className="card-body"
              >
                <CardTitle>
                  <ForwardRef
                    className="card-title"
                  >
                    <div
                      className="card-title h5"
                    >
                      Jenna
                    </div>
                  </ForwardRef>
                </CardTitle>
                <CardText>
                  <p
                    className="card-text"
                  >
                    1 Queen
                    2
                  </p>
                </CardText>
                <Button
                  active={false}
                  disabled={false}
                  type="button"
                  variant="primary"
                >
                  <button
                    className="btn btn-primary"
                    disabled={false}
                    type="button"
                  >
                    More Info
                  </button>
                </Button>
              </div>
            </CardBody>
          </div>
        </Card>
      </div>
    </Restaurant>
    <Restaurant
      address="3 Front"
      key="Troy1"
      name="Troy"
      price={2}
    >
      <div
        className="Restaurant"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <Card
          body={false}
          style={
            Object {
              "width": "18rem",
            }
          }
        >
          <div
            className="card"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <CardImg
              variant="top"
            >
              <img
                className="card-img-top"
              />
            </CardImg>
            <CardBody>
              <div
                className="card-body"
              >
                <CardTitle>
                  <ForwardRef
                    className="card-title"
                  >
                    <div
                      className="card-title h5"
                    >
                      Troy
                    </div>
                  </ForwardRef>
                </CardTitle>
                <CardText>
                  <p
                    className="card-text"
                  >
                    3 Front
                    2
                  </p>
                </CardText>
                <Button
                  active={false}
                  disabled={false}
                  type="button"
                  variant="primary"
                >
                  <button
                    className="btn btn-primary"
                    disabled={false}
                    type="button"
                  >
                    More Info
                  </button>
                </Button>
              </div>
            </CardBody>
          </div>
        </Card>
      </div>
    </Restaurant>
  </div>
</SearchList>
`;
